pkg/storage/kubernetes.go

```diff
@@ -227,3 +239,197 @@ func (t *temporaryError) Temporary() bool {
 type temporary interface {
 	Temporary() bool
 }
+
+// newLeaderElector creates a new leaderelection.LeaderElector and associated
+// channels by which to observe elections and depositions.
+func newLeaderElector(clientset *kubernetes.Clientset, namespace string, podNamespace string, podID string) (*leaderelection.LeaderElector, chan struct{}, chan struct{}) {
+	//log.WithField("context", "leaderelection")
+	// leaderOK will block gRPC startup until it's closed.
+	leaderOK := make(chan struct{})
+	// deposed is closed by the leader election callback when
+	// we are deposed as leader so that we can clean up.
+	deposed := make(chan struct{})
+
+	var rl = &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      "whereabouts",
```

Comment:
you could potentially use the name of the pool here, so that allocations in other pools are not also blocked

Reply:
I suppose we should use leaselock to lock whole whereabouts, not per ip pool basis because Kubernetes/etcd does not support fine grain lock (because etcd is not DB, just KVS).